src/univariate/continuous/normal.jl

```diff
@@ -105,14 +105,14 @@ rand(rng::AbstractRNG, d::Normal{T}) where {T} = d.μ + d.σ * randn(rng, float(
 
 #### Fitting
 
-struct NormalStats <: SufficientStats
-    s::Float64    # (weighted) sum of x
-    m::Float64    # (weighted) mean of x
-    s2::Float64   # (weighted) sum of (x - μ)^2
-    tw::Float64    # total sample weight
+struct NormalStats{T} <: SufficientStats
+    s::T    # (weighted) sum of x
+    m::T    # (weighted) mean of x
+    s2::T   # (weighted) sum of (x - μ)^2
+    tw::T    # total sample weight
 end
 
-function suffstats(::Type{<:Normal}, x::AbstractArray{T}) where T<:Real
+function suffstats(::Type{Normal{T}}, x::AbstractArray{<:Real}) where {T<:Real}
```

Comment:
I don't think the type parameter of `Normal` should matter here. The sufficient statistics are very much data-dependent, and it might even cause surprising errors if the type parameter is not "wide" enough (since in general the sufficient statistics don't have to be of the same type as the elements of `x`). `suffstats` is just an intermediate step in the `fit` pipeline, and the type of the resulting distribution is only determined when calling `fit`, either with data or the sufficient statistics.

Reply:
```suggestion
function suffstats(::Type{<:Normal}, x::AbstractArray{<:Real})
```